src/Features/Core/Portable/SolutionCrawler/ISolutionCrawlerProgressReporter.cs

```diff
@@ -18,11 +19,28 @@ internal interface ISolutionCrawlerProgressReporter
         /// <summary>
         /// Raised when solution crawler progress changed
         /// 
-        /// bool indicates whether progress is started or stopped
-        /// 
         /// Notifications for this event are serialized to preserve order. 
         /// However, individual event notifications may occur on any thread.
         /// </summary>
-        event EventHandler<bool> ProgressChanged;
+        event EventHandler<ProgressData> ProgressChanged;
+    }
+
+    internal struct ProgressData
+    {
+        public Status Type { get; }
+        public string FilePath { get; }
+
+        public ProgressData(Status type, string filePath)
```

Comment:
Looking at the call sites below it seems the `filePath` can be null. Can you add comments and/or call it `FilePathOpt`, and add comments when null is expected?

Reply:
okay.